### PR TITLE
Save events

### PR DESCRIPTION
**Pull request by FabrizioCaldarelli**
I need to split save events in "insert" and "update" because needed from some classes, for example TimestampBehavior.

### CODE REVIEW

File: src/ActiveRecordSaveEvent.php [hunk 1]
```diff
@@ -59,7 +69,7 @@ public function __construct(string $name, bool $insert, array $changedAttributes
      */
     public static function before(bool $insert): self
```
Comment (FabrizioCaldarelli) [inline src/ActiveRecordSaveEvent.php]: Are two method preferred instead one?

I mean beforeInsert and beforeUpdate

Comment (samdark) [reply src/ActiveRecordSaveEvent.php]: Yes, two methods are better.

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by kids-return** [comment]
Are there any good corrections or suggestions for the event of `yii2-queue`?
```php
    public function push($job)
    {
        $event = new PushEvent([
            'job' => $job,
            'ttr' => $this->pushTtr ?: (
                $job instanceof RetryableJobInterface
                    ? $job->getTtr()
                    : $this->ttr
            ),
            'delay' => $this->pushDelay ?: 0,
            'priority' => $this->pushPriority,
        ]);
        $this->pushTtr = null;
        $this->pushDelay = null;
        $this->pushPriority = null;
        $this->trigger(self::EVENT_BEFORE_PUSH, $event);
        if ($event->handled) {
            return null;
        }
        if ($this->strictJobType && !($event->job instanceof JobInterface)) {
            throw new InvalidArgumentException('Job must be instance of JobInterface.');
        }
        $message = $this->serializer->serialize($event->job);
        $event->id = $this->pushMessage($message, $event->ttr, $event->delay, $event->priority);
        $this->trigger(self::EVENT_AFTER_PUSH, $event);
        return $event->id;
    }
```

**Comment by kids-return** [comment]
Is this correct?
```php
    public function push($job)
    {
        $beforeEvent = JobEvent::beforePush($job, $this->pushTtr ?: (
        $job instanceof RetryableJobInterface
            ? $job->getTtr()
            : $this->ttr
        ), $this->pushDelay ?: 0, $this->pushPriority);


        $afterEvent = JobEvent::afterPush($job, $this->pushTtr ?: (
        $job instanceof RetryableJobInterface
            ? $job->getTtr()
            : $this->ttr
        ), $this->pushDelay ?: 0, $this->pushPriority);

        $this->pushTtr = null;
        $this->pushDelay = null;
        $this->pushPriority = null;

        $this->trigger($beforeEvent);
        if ($beforeEvent->isPropagationStopped()) {
            return null;
        }

        if ($this->strictJobType && !($beforeEvent->job instanceof JobInterface)) {
            throw new InvalidArgumentException('Job must be instance of JobInterface.');
        }

        $message = $this->serializer->serialize($beforeEvent->job);
        $afterEvent->id = $this->pushMessage($message, $beforeEvent->ttr, $beforeEvent->delay, $beforeEvent->priority);
        $this->trigger($afterEvent);

        return $afterEvent->id;
    }
```

**Comment by samdark** [comment]
@kids-return queue isn't related to active record.

**Pull request merged**